src/ansys/mapdl/core/mapdl_inprocess.py

```diff
@@ -0,0 +1,65 @@
+# Copyright (C) 2024 ANSYS, Inc. and/or its affiliates.
+# SPDX-License-Identifier: MIT
+#
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+from typing import Optional, Protocol
+
+from ansys.mapdl.core.mapdl import MapdlBase
+
+
+class _Backend(Protocol):
+    def run_command(self) -> str: ...
+
+
+class MapdlInProcess(MapdlBase):
+    def __init__(self, backend: _Backend):
+        super().__init__(
+            loglevel="WARNING", use_vtk=False, log_apdl=None, print_com=False
+        )
+        self._backend = backend
```

Comment:
This name is quite generic.. I would expect this property in the other subclasses, or this being more explicit (for instance ``self._in_memory_backend``).